app/models/pipeline_run.rb

```diff
@@ -542,16 +542,31 @@ def db_load_contigs(contig2taxid)
     taxon_lineage_map = {}
     TaxonLineage.where(taxid: taxid_list.uniq).order(:id).each { |t| taxon_lineage_map[t.taxid.to_i] = t.to_a }
 
+    # A lambda allows us to access variables in the enclosing scope, such as contig2taxid.
+    get_contig_hash = lambda do |header, sequence|
+      read_count = contig_stats_json[header] || 0
+      lineage_json = get_lineage_json(contig2taxid[header], taxon_lineage_map)
+      species_taxid_nt = lineage_json.dig("NT", 0) || nil
+      species_taxid_nr = lineage_json.dig("NR", 0) || nil
+      genus_taxid_nt = lineage_json.dig("NT", 1) || nil
+      genus_taxid_nr = lineage_json.dig("NR", 1) || nil
+
+      {
+        name: header, sequence: sequence, read_count: read_count, lineage_json: lineage_json.to_json,
+        species_taxid_nt: species_taxid_nt, species_taxid_nr: species_taxid_nr,
+        genus_taxid_nt: genus_taxid_nt, genus_taxid_nr: genus_taxid_nr,
+      }
+    end
+
     File.open(contig_fasta, 'r') do |cf|
```

Comment:
maybe out of scope for this PR, but it seems like a fasta file parser should be in a util module. 

also, I'm surprised the pipeline does not do the parsing and generate the structured info we need here in a json file